### PR TITLE
The Metal tier's second fold round: the row chains share their adapters, the mv shells lay their lane constants out once, the iq4 mv pair and the double-buffered mul_mm kernel are one template each

### PR DESCRIPTION
**Pull request by borisbat**
**Why.** The dupe audit of the last fold round named four sets that were still two copies: five verify/batch adapters with identical bodies, a verify site that was the batch ladder with its arms off, nine `lane_setup` overrides carrying three maps, and the iq4xs/iq4nl mv pair one scale line apart; the standalone sweep added the K4/K5 and K6 double-buffered mul_mm kernels, identical over their decoders.

**What changes.**
- The verify and batch row chains bind one generic adapter per slot (`rms_rows`, `add_rows`, `add_rms_rows`, `moe_args_rows`, `total_rows`), and the verify's weight site is the batch ladder under `rows_tier`, a tier with every arm off.
- A `MetalKqLaneMapsT` base both mv shells derive from carries the 2/3-bit, 4/5-bit and 6-bit lane maps behind one `LANES` constant; a stamp names its map, no override remains.
- `MetalKqMvIq4T` and `MetalKqMvB8Iq4T` carry the iq4 mv kernels with the scale source as their one axis (`IQ4NL`); stamps keep every dispatch name.
- `MetalKqMulMmDbShellT` carries the double-buffered dev-W mul_mm kernel over an abstract `stage_chunk`; the K4/K5 and K6 templates keep only their decoders.

**Observable behavior.**
- Every emitted kernel's AIR is byte-identical to master's (299 of 299), so no served number moves.
- The unfused add+rms pin (`set_metal_batch_addrms_unfused`) now steers the same-slab verify chain as well as the batch chain; before it steered the batch only.
- A row-total request for a uniform whose twin the step never built: a null bind -> a panic naming the uniform.
- The generated globals of the B8 iq4 kernels and the K6 double-buffer stamps are spelled by their stamps and the shared kernel (`MetalKqMvB8Iq4xs_metal_kq_mvb8_iq4_msl`, `MetalKqMulMmK6THDb_metal_kq_mulmm_db_msl`); the two kernel tests are updated.

**Where to look.** `rows_tier` and `total_rows` in `dasllama_metal_decode.das`, the `LANES` maps and the `IQ4NL` `static_if` in `dasllama_metal_kernels.das`, and the `MetalKqMulMmDbShellT` head in `dasllama_metal_prefill.das`.

<details>
<summary>Validation, claims, ledger</summary>

### Validation
- Emitted-kernel identity: every kernel's MSL dumped and compiled to AIR with `xcrun metal -O2 -c`, disassembled with `metal-opt -S -discard-value-names`, normalized and diffed against master's set after each fold (out-of-tree instrument `~/Work/dupe-sweeps/2026-09-11-metal/tools/msl_compare.sh`, before = master 07ef806f2): 299 identical, 0 moved, 0 changed. With identical machine code nothing was raced.
- Model suites on the M5 Max through `tests/run.das` (`-jit` children): `--suite kernels --arm kernels`, `--suite mtp` on all seven arms (the 0.8b MTP round drafts 48 rows served, the Llama-1B same-slab verify serves 12 steps plain and 12 under the pin), `--suite decode --arm batch,batchB8-kq,arm1-basic,arm10-kq,arm12-dim`, `--suite prefill --arm base,kq,mm-tail`, `--suite image --arm mtower,gemma4a,gemma4uv`, `--suite coverage`, and the changed-file rail over every area (122 files; the two reds are `test_vulkan_dec_tail` and `test_vulkan_tier`, red on master on this box).
- Negative control from the main tree: a value change in the 6-bit lane map reds the k6 mv cells of `test_metal_gemv_kernels.das` and nothing else; restored.
- `preflight --full` ran once, all lanes green (20 passed, 5 skipped for no C++ or utils change).
- The kernel-unit gates for the renamed symbols: `test_metal_gemv_kernels.das` and `test_metal_gemm_kernels.das` fail to compile against the old names and pass against the new.

### Claims - stated, not tested
- The `total_rows` null-twin panic is unreachable today: the batch step gate declines the gated-Q and second-width models whose uniforms have no twin.
- The large-tier MTP fixtures (Qwen3.6-27B, 35B-A3B, Qwen3.8 split head, gemma-4-26B) did not run; they need `DASLLAMA_PARITY_FULL=1` and their served chains are the ones the Llama-1B and 0.8b fixtures exercise.

### Not done
- The iq4 templates' 8-line weight-decode loop is written in both the B2/B4 and the B8 template; a free helper would take a fixed-array parameter the MSL emitter lowers as threadgroup memory, so it stays.
- The `_t` tensor family's mul_mm kernels carry inline copies of the double-buffer decoders that `stage_chunk` now names; a follow-up fold.
- `MT` on the mul_mm shells reads as "M tile"; renaming it across every stamp is a separate change.

</details>

### CODE REVIEW

File: modules/dasLLAMA/dasllama/dasllama_metal_decode.das [hunk 1]
```diff
@@ -1344,70 +1343,61 @@ def private metal_mtp_verify_ok(t : Model; s : Session) : bool {
...
-    enc_rms_b(enc, bs, soff, bw, bd, e.r.u_dim, e.r.u_eps, e.r.nrows)
+def private total_rows(e : auto(ET); u : MetalBuffer?; n : int64) : tuple<b : MetalBuffer?; n : int64> {
+    let r & = unsafe(*e.r)
+    if (u != r.u_dim && u != r.u_hidden && u != r.u_qd && u != r.u_moe_knfe) {
+        panic("row chain: a row-total is asked for a uniform with no nrows-scaled twin")
+    }
+    let bt = u == r.u_dim ? r.u_total_dim : (u == r.u_hidden ? r.u_total_h : (u == r.u_qd ? r.u_total_qd : r.u_moe_tot))
+    if (bt == null) {
+        panic("row chain: the row-total twin of this uniform was not built for this step (the step gate declines the models that would ask)")
+    }
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLAMA/dasllama/dasllama_metal_decode.das:1380]: Accepted: the panic now names the slot (`u_dim` / `u_hidden` / `u_qd` / `u_moe_knfe`) whose twin was not built (fc4ff4349).

File: modules/dasLLAMA/dasllama/dasllama_metal_kernels.das [hunk 2]
```diff
@@ -8123,11 +8123,39 @@ class MetalKqGemvK2 {
     }
 }
 
+
+//! LANES picks the map lane_setup lays into lc[] - 23 = k2/k3, 45 = k4/k5, 6 = k6, 0 = none; an mv_superblock reads the slots back by position
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLAMA/dasllama/dasllama_metal_kernels.das]: Accepted: all three are plain `//` comments now (fc4ff4349).

File: modules/dasLLAMA/tests/test_metal_mtp_parity.das [hunk 3]
```diff
@@ -727,6 +727,17 @@ def test_metal_mtp_parity_qwen38_split_head(t : T?) {
 // A head-less fixture: only the batch-rail verify arm runs. gemma4's router amplifies any
 // float-class difference to ~2.5 logits/step (the support-matrix row's finding), so its bar is
 // the row's, not the 0.5 one.
+//! one row through the batch driver, plain and under the unfused add+rms pin - the row chains'
+//! shared adapter takes the add + rms pair instead of the fused kernel
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLAMA/tests/test_metal_mtp_parity.das]: Accepted: both lines are plain `//` comments now (fc4ff4349).
